docs/modules/ROOT/pages/index.adoc

```diff
@@ -38,4 +64,308 @@ include::includes/quarkus-solace-extension-outgoing.adoc[leveloffset=+1, opts=op
+
+A minimal configuration for the Solace connector with an incoming channel looks like the following:
+
+The following lines of configuration assumes that a exclusive queue is already provisioned on the broker
+[source,properties]
+----
+quarkus.solace.host=tcp://localhost:55555
+quarkus.solace.vpn=default
+quarkus.solace.authentication.basic.username=basic
+quarkus.solace.authentication.basic.password=basic
+
+mp.messaging.incoming.temperatures.connector=quarkus-solace
+mp.messaging.incoming.temperatures.consumer.queue.name=temperatures
+----
+
+The extension also supports provisioning queues and subscriptions on broker given that the user has role access to create queues with subscriptions. Configuration is as follows
+
+[source,properties]
+----
+quarkus.solace.host=tcp://localhost:55555
+quarkus.solace.vpn=default
+quarkus.solace.authentication.basic.username=basic
+quarkus.solace.authentication.basic.password=basic
+
+mp.messaging.incoming.temperatures.connector=quarkus-solace
+mp.messaging.incoming.temperatures.consumer.queue.missing-resource-creation-strategy=create-on-start
+mp.messaging.incoming.temperatures.consumer.queue.add-additional-subscriptions=true
+mp.messaging.incoming.temperatures.consumer.queue.subscriptions=hello/foobar
+----
+
+1. When running in dev mode or tests dev services will automatically start a Solace PubSub+ broker and if broker configuration details are not provided the extension automatically picks up the details of broker started by dev services.
+
+2. If `consumer.queue.name` property is not specified, channel name will be used as queue name.
+
+[[receiving-messages-from-solace]]
+== Receiving messages from Solace
+
+Using the previous configuration, Quarkus application can receive message in several possible ways.
+
+__Direct Payload__
+[source,java]
+----
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+import jakarta.enterprise.context.ApplicationScoped;
+@ApplicationScoped
+public class TemperaturesConsumer {
+    @Incoming("temperatures")
+    public void consume(Double temperature) {
+        // process.
+    }
+}
+----
+
+__Message__
+[source,java]
+----
+@ApplicationScoped
+public class TemperaturesConsumer {
+    @Incoming("temperatures")
+    public CompletionStage<Void> consume(Message<Double> msg) {
+        // access record metadata
+        SolaceInboundMetadata metadata = msg.getMetadata(SolaceInboundMetadata.class).orElseThrow();
+        // process the message payload.
+        Double temperature = msg.getPayload();
+        // Acknowledge the incoming message
+        return msg.ack();
+    }
+}
+----
+
+__SolaceInboundMessage__ This is a wrapper to incoming Inbound Message from Solace Messaging API
```

Comment:
We are starting to discourage usage of custom message type like `SolaceInboundMessage`, as it doesn't allow any kind of interception in message stream.